addOns/client/src/main/java/org/zaproxy/addon/client/spider/ClientSpider.java

```diff
@@ -111,7 +116,36 @@ public void start() {
                         new ClientSpiderThreadFactory(
                                 "ZAP-ClientSpiderThreadPool-" + id + "-thread-"));
 
+        List<String> unvisitedUrls = getUnvisitedUrls();
+
         addTask(targetUrl, options.getInitialLoadTimeInSecs());
+
+        // Add all of the known but unvisited URLs otherwise these will get ignored
+        unvisitedUrls.forEach(url -> addTask(url, options.getInitialLoadTimeInSecs()));
```

Comment:
Shouldn't this be a collection that's shared across spiders/threads. Won't they only be "unvisited" for the current browser, though not necessarily for _n_ running spiders?

Reply:
They are only unvisited at the moment that the spider is started. they could be visited at any point after that, so I dont think anything needs to be shared.